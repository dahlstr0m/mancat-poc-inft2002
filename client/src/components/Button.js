// @flow

import * as React from 'react';
import { Component } from 'react-simplified';

/**
 * Renders a success button using Bootstrap styles.
 */
class ButtonSuccess extends Component<{
  onClick: () => mixed,
  small?: boolean,
  children?: React.Node,
}> {
  render() {
    return (
      <button
        type="button"
        className={'btn btn-success' + (this.props.small ? ' btn-sm py-0' : '')}
        onClick={this.props.onClick}
      >
        {this.props.children}
      </button>
    );
  }
}

/**
 * Renders a danger button using Bootstrap styles.
 */
class ButtonDanger extends Component<{
  onClick: () => mixed,
  small?: boolean,
  children?: React.Node,
}> {
  render() {
    return (
      <button
        type="button"
        className={'btn btn-danger' + (this.props.small ? ' btn-sm py-0' : '')}
        onClick={this.props.onClick}
      >
        {this.props.children}
      </button>
    );
  }
}

/**
 * Renders a light button using Bootstrap styles.
 */
class ButtonLight extends Component<{
  onClick: () => mixed,
  small?: boolean,
  children?: React.Node,
}> {
  render() {
    return (
      <button
        type="button"
        className={'btn btn-light' + (this.props.small ? ' btn-sm py-0' : '')}
        onClick={this.props.onClick}
      >
        {this.props.children}
      </button>
    );
  }
}

/**
 * Renders a button using Bootstrap styles.
 */
export default class Button {
  static Success = ButtonSuccess;
  static Danger = ButtonDanger;
  static Light = ButtonLight;
}
