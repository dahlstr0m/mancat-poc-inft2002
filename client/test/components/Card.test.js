// @flow

import * as React from 'react';
import { shallow } from 'enzyme';

// Components
import {
  Card,
  TitleCard,
  CardBody,
  CardImage,
  CardImageEffect,
  CardColumn,
  CardGrid,
} from '../../src/components/Card';
import Button from '../../src/components/Button';

// Tests
describe('Card components tests', () => {
  test('Card component draws correctly', () => {
    const wrapper = shallow(<Card title="Title" textRight></Card>);

    expect(wrapper).toMatchSnapshot();
  });

  test('TitleCard component draws correctly', () => {
    const wrapper = shallow(<TitleCard img="src" button buttonText="button"></TitleCard>);

    expect(wrapper).toMatchSnapshot();
  });

  test('CardBody draws correctly', () => {
    const wrapper = shallow(<CardBody title="title"></CardBody>);

    expect(wrapper).toMatchSnapshot();
  });

  test('CardImage draws correctly', () => {
    const wrapper = shallow(
      // $FlowExpectedError
      <CardImage title="title" img="src" imgAlt="alt" button buttonText="button"></CardImage>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('CardImageEffect draws correctly', () => {
    const wrapper = shallow(
      <CardImageEffect title="title" img="src" imgAlt="alt" link="/link"></CardImageEffect>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('CardColumn draws correctly', () => {
    const wrapper = shallow(<CardColumn></CardColumn>);

    expect(wrapper).toMatchSnapshot();
  });

  test('CardGrid draws correctly', () => {
    const wrapper = shallow(
      <CardGrid columns={1} columnsSm={2} columnsMd={3} columnsLg={4}></CardGrid>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
