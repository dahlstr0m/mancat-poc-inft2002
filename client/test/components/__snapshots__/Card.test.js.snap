// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card components tests Card component draws correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Title
    </h5>
    <div
      className="card-text text-right"
    />
  </div>
</div>
`;

exports[`Card components tests CardBody draws correctly 1`] = `
<div
  className="card-body"
>
  <h5
    className="card-title"
  >
    title
  </h5>
  <div
    className="card-text"
  />
</div>
`;

exports[`Card components tests CardColumn draws correctly 1`] = `
<div
  className="col mb-4"
/>
`;

exports[`Card components tests CardGrid draws correctly 1`] = `
<div
  className="row row-cols-1 row-cols-sm-2 row-cols-md-3 row-cols-lg-4"
/>
`;

exports[`Card components tests CardImage draws correctly 1`] = `
<div
  className="card h-100"
>
  <div
    className="card-body"
  >
    <img
      alt="alt"
      className="card-img-top rounded mx-auto d-block"
      src="src"
    />
    <hr />
    <h5
      className="card-title mx-auto d-block"
    >
      title
    </h5>
    <div
      className="card-text text-center"
    />
    <div
      className="text-center"
    >
      <ButtonDanger
        onClick={[Function]}
      >
        button
      </ButtonDanger>
    </div>
  </div>
</div>
`;

exports[`Card components tests CardImageEffect draws correctly 1`] = `
<div
  className="card hover-shadow"
>
  <div
    className="card-body"
  >
    <NavLink
      to="/link"
    >
      <img
        alt="alt"
        className="rounded mx-auto d-block"
        src="src"
      />
    </NavLink>
  </div>
</div>
`;

exports[`Card components tests TitleCard component draws correctly 1`] = `
<div
  className="card"
>
  <h1>
    <img
      alt="-No img-"
      height={140}
      src="src"
      style={
        Object {
          "marginRight": "15px",
        }
      }
      width={140}
    />
  </h1>
  <ButtonLight
    onClick={[Function]}
  >
    button
  </ButtonLight>
</div>
`;
